include/ginkgo/core/log/solver_debug.hpp

```diff
@@ -0,0 +1,73 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef GKO_PUBLIC_CORE_LOG_SOLVER_DEBUG_HPP_
+#define GKO_PUBLIC_CORE_LOG_SOLVER_DEBUG_HPP_
+
+
+#include <iosfwd>
+#include <memory>
+
+
+#include <ginkgo/core/log/logger.hpp>
+
+
+namespace gko {
+namespace log {
+
+
+/**
+ * This Logger prints the value of all scalar values stored internally by the
+ * solver after each iteration. If the solver is applied to multiple right-hand
+ * sides, only the first right-hand side gets printed.
+ */
+class SolverDebug : public Logger {
```

Comment:
Pinging this again, I'm still unhappy with the name. It's rather non-descriptive, so users will have no idea what this is doing just from the name.

Reply:
How would you describe what this logger is used for? My first thought was "It's for debugging solvers", so the name made sense. But from a user perspective, it's maybe slightly unintuitive. OTOH, will users need to use this regularly?

Reply:
It's for debugging solvers in a very specific sense, by printing the internal state after every iteration. That can be reflected in the name. Just 'debug' can mean anything and nothing.
Regarding usage: it could also be used to achieve the same thing as in #1517, so perhaps users will use it for that.

Reply:
How about `SolverProgress`?

Reply:
TBH, I kinda warmed up to my first suggestion. `SolverProgress` could easily be understood to mean just the residual norm, so I think referencing the state might be useful.